sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreSuite.scala

```diff
@@ -0,0 +1,152 @@
+    val kv = iter.next()
+
+    // Verify the version encoded in first byte of the key and value byte arrays
+    assert(Platform.getByte(kv.key, Platform.BYTE_ARRAY_OFFSET) === STATE_ENCODING_VERSION)
+    assert(Platform.getByte(kv.value, Platform.BYTE_ARRAY_OFFSET) === STATE_ENCODING_VERSION)
+  }
+
+  test("RocksDB confs are passed correctly from SparkSession to db instance") {
+    val sparkConf = new SparkConf().setMaster("local").setAppName(this.getClass.getSimpleName)
+    withSparkSession(SparkSession.builder.config(sparkConf).getOrCreate()) { spark =>
+      // Set the session confs that should be passed into RocksDB
+      val testConfs = Seq(
+        ("spark.sql.streaming.stateStore.providerClass",
+          classOf[RocksDBStateStoreProvider].getName),
+        (RocksDBConf.ROCKSDB_CONF_NAME_PREFIX + ".compactOnCommit", "true"),
+        (RocksDBConf.ROCKSDB_CONF_NAME_PREFIX + ".lockAcquireTimeoutMs", "10")
+      )
+      testConfs.foreach { case (k, v) => spark.conf.set(k, v) }
+
+      // Prepare test objects for running task on state store
+      val testRDD = spark.sparkContext.makeRDD[String](Seq("a"), 1)
+      val testSchema = StructType(Seq(StructField("key", StringType, true)))
+      val testStateInfo = StatefulOperatorStateInfo(
+        checkpointLocation = Utils.createTempDir().getAbsolutePath,
+        queryRunId = UUID.randomUUID, operatorId = 0, storeVersion = 0, numPartitions = 5)
+
+      // Create state store in a task and get the RocksDBConf from the instantiated RocksDB instance
+      val rocksDBConfInTask: RocksDBConf = testRDD.mapPartitionsWithStateStore[RocksDBConf](
+        spark.sqlContext, testStateInfo, testSchema, testSchema, None) {
+          (store: StateStore, _: Iterator[String]) =>
+            // Use reflection to get RockDB instance
+            val dbInstanceMethod =
+              store.getClass.getMethods.filter(_.getName.contains("dbInstance")).head
+            Iterator(dbInstanceMethod.invoke(store).asInstanceOf[RocksDB].conf)
+        }.collect().head
+
+      // Verify the confs are same as those configured in the session conf
+      assert(rocksDBConfInTask.compactOnCommit == true)
+      assert(rocksDBConfInTask.lockAcquireTimeoutMs == 10L)
+    }
+  }
+
+  test("rocksdb file manager metrics exposed") {
+    import RocksDBStateStoreProvider._
+    def getCustomMetric(metrics: StateStoreMetrics, customMetric: StateStoreCustomMetric): Long = {
+      val metricPair = metrics.customMetrics.find(_._1.name == customMetric.name)
+      assert(metricPair.isDefined)
+      metricPair.get._2
+    }
+
+    val provider = newStoreProvider()
+    val store = provider.getStore(0)
+    // Verify state after updating
+    put(store, "a", 1)
+    assert(get(store, "a") === Some(1))
+    assert(store.commit() === 1)
+    assert(store.hasCommitted)
+    val storeMetrics = store.metrics
+    assert(storeMetrics.numKeys === 1)
+    assert(getCustomMetric(storeMetrics, CUSTOM_METRIC_FILES_COPIED) == 1L)
```

Comment:
nit: I guess we can loose the condition via `> 0L`. "The number of files is 1" doesn't seem to something we need to check.

Reply:
Agree, will update in the next commit.